src/main/java/algorithms/curated170/medium/DesignTicTacToe.java

```diff
@@ -0,0 +1,34 @@
+package algorithms.curated170.medium;
+
+public class DesignTicTacToe {
+
+    public class TicTacToe {
+        int[][] countRow;
+        int[][] countCol;
+        int[] countDia1;
+        int[] countDia2;
+        int length;
+   
+        public TicTacToe(int n) {
+            length = n;
+            countRow = new int[n][3];
+            countCol = new int[n][3];
+            countDia1 = new int[3];
+            countDia2 = new int[3];
+        }
+   
+        public int move(int row, int col, int player) {
+          
+         return ((++countRow[row][player] == length) 
+             ||  (++countCol[col][player] == length)
+             ||  (row == col && ++countDia1[player] == length)
+             ||  (row + col + 1 == length &&  ++countDia2[player] == length)
```

Comment:
I have added parenthesis around these statements to make them look a clearer. It will also speed up the code by a really tiny, almost negligible fraction, but in diagonals, the method won't try to look up the array's value if row is not equal to the column or the other possibility.